### DISCUSSION TITLE
Difficulty with background cutting.

### DISCUSSION THREAD

**Discussion by Miyuutsu** [Extraltodeus/multi-subject-render#15] [General]
I seem to be having a lot of difficulty with background cutting regardless of the prompts or settings being used. I can't get a middle ground between having enough background removed to not need high enough steps/denoise that all my characters become the same even when vastly different or otherwise having it look like a pasted on stone.

The `bad_prompt:0.8` is from the bad_prompt embedding, it's basically an embedding trained on default NAI negative prompt. I can not provide the model I am using however it is based upon ChiNAI (Anything V3). Despite this I seem to have this problem across most models I've tried.

Prompt: `masterpiece, best quality, bedroom`
Negative prompt: `(bad_prompt:0.8),artist name,missing anatomy,extra anatomy,missing toe,extra toe,extra fingers,thick lips,sweat,disfigured,deformed,malformed,mutant,monstrous,ugly,gross,disgusting,man,male,boy,old,fat,obese,out of frame,poorly drawn,extra limbs,missing limbs,duplicate,morbid,mutated,poorly drawn,long neck,long body,poorly drawn`

Sampling: `28`
Sampler: `Euler a`
CFG Scale: `9`
Seed: `313617661`
Size: 512x512

Foreground prompt: `masterpiece,best quality,(wide angle),SFW,full body,portrait,medium hair,pink hair,cat ears,cat tail,red eyes,blue dress,sitting,:d,(bokeh),(f1.8:1.2)`

Foreground images: `1`
Foreground steps: `28`
Foreground CFG Scale: `12.5`
Foreground Sampler: `Euler a`
Foreground Width: `64`
Foreground Height: `896`

Final Blend Prompt: `masterpiece, best quality, SFW, girl, cat ears, cat tail, bedroom`
Blend Steps: `70`
Blend CFG Scale: `7.5`
Blend Denoising Strength: `.42`
Blend Sampler: `DPM++ 2M Karras`
Blend Width: `1536`
Blend Height: `896`

Foreground Depth Cut Threshold: `70`

I'm only doing one foreground image for the sake of simplicity in identifying the issue, I don't really expect that it line up perfectly in this case. Images in order are background, foreground, blended.

![00040-313617661-masterpiece, best quality, bedroom](https://user-images.githubusercontent.com/20072970/204075629-1f340e64-3cd6-4b81-aa3e-df4acffd5ded.png)
![00041-313617661-masterpiece,best quality,(wide angle),SFW,full body,portrait,medium hair,pink hair,cat ears,cat tail,red eyes,blue dress,sitting](https://user-images.githubusercontent.com/20072970/204075630-ba2db361-b92a-44cb-b458-ca60a9dfa99f.png)
![00042-313617661-masterpiece, best quality, SFW, girl, cat ears, cat tail, bedroom](https://user-images.githubusercontent.com/20072970/204075634-f2fb8b1d-4ca1-4907-9a12-dab63d2780c8.png)

There's quite a lot of background that does not get cut out in this case. If I take the Foreground Depth Cut Threshold to 160 instead of 70 I can finally get most of my foreground character on her own.
 
![00055-313617661-masterpiece, best quality, SFW, girl, cat ears, cat tail, bedroom](https://user-images.githubusercontent.com/20072970/204076189-ed29e236-dfa3-459e-8969-7426a18b9736.png)

At this point however it doesn't seem to be enough to simply increase blending steps or denoising strength, because although going higher can help with this going higher also causes my characters to all become very similar if I'm using more than one. Increasing the Foreground Depth Cut Thereshold further is of no benefit because I eventually reach a point where I'm also cutting the character out. This is at 170, there's still background but it's begun to cut the character.

![00056-313617661-masterpiece, best quality, SFW, girl, cat ears, cat tail, bedroom](https://user-images.githubusercontent.com/20072970/204076199-da626f4d-38ce-44d0-9e92-06c1943a1653.png)

I've tried all the different samplers for each step as well as Clip Skip 1 and 2 for each step. I've tried keeping everything at 512px except for final blend width in case it was an issue of higher resolution outputs. Changing the Blend CFG Scale can occasionally have some stronger impact. I've tried variations to the bokeh and f1.8 that are suggested and occasionally I can get better cuts without them entirely but rarely so. Seems most of my best cuts are when I assign a similar background to the foreground prompt and it comes out close enough to the background to be denoised together sensibly but that's next to impossible to perform reliably with more than a single foreground. There may be some typos here that are inaccurate to anything actually done since I'm exhausted and spent an hour writing this so please bear that in mind. Any advice?

**Comment by Extraltodeus** [comment]
Quickly answering right now cause I'm on my phone but I've read it all and I would use the second image then raise the denoising strength and steps in the final blend so the leftovers are reworked into something else. Also trying to find a prompt where this gray ground does not appear would of course help.

**Reply by Miyuutsu** [reply]
Thanks for the response! I'm on mobile right now as well, probably won't be able to test for a few days. Yeah that gray ground bit is weird, I believe that's probably a side effect of my model but it seems like my choices with that prompt specifically tend to be either ground of some kind of a green bokeh that doesn't get handled well. 

Is it normal at all to have output that requires such a high depth cut threshold, or is it possible that the readme has a typo and you meant 150-160 instead?

**Reply by Extraltodeus** [reply]
it REALLY depenmds on the subject/background. It's all an analysis made by MiDaS. I encourage you to save the depthmaps while generating so you can get a better idea of what is going on too!

**Reply by Miyuutsu** [reply]
I can't say that I'm familiar with MiDaS. I'll need to look into that!
